share/spice/whois/whois.js

```diff
@@ -66,8 +73,10 @@
         }
 
         // check we have 2 nameserver hostnames
-        if(nsObj && nsObj.hostNames[0] && nsObj.hostNames[1]) {
-            var nameServers = [nsObj.hostNames[0].toLowerCase(),nsObj.hostNames[1].toLowerCase()].join(' ');
+        if(nsObj.hostNames) {
+            if(nsObj.hostNames.length >= 2) {
+                var nameServers = [nsObj.hostNames[0].toLowerCase(),nsObj.hostNames[1].toLowerCase()].join(' ');
```

Comment:
Stop nitpicking @jagtalon :laughing: Unfortunately if you try checking if  `nsObj.hostNames[0-1]` exists and it doesn't you'll end up with an `undefined` typeError so we first check the length. :+1: